File: backend/app/data/schemas/customer_investment.py
from pydantic import BaseModel, Field
from marshmallow import Schema, fields
from .investment import InvestmentRes

class CustomerInvestmentCreateReq(BaseModel):
    customer_id: int = Field(...)
    title: str = Field(...)
    amount: int = Field(...)
    investment_id: int = Field(...)

class CustomerInvestmentRes(Schema):
    id = fields.Int()
    title = fields.Str()
    amount = fields.Int()
    appreciation = fields.Int()
    created_at = fields.DateTime()
    investment = fields.Nested(InvestmentRes)
    
class CustomerInvestmentResOmitInvestment(Schema):
    id = fields.Int()
    title = fields.Str()
    amount = fields.Int()
    created_at = fields.DateTime()

class CustomerInvestmentWithdrawReq(BaseModel):
    id: int = Field(...)
    amount: int = Field(...)

File: frontend/src/types/customer_investment.ts
import { InvestmentRes } from './investment';

export interface CustomerInvestmentCreateReq {
    customer_id: number;
    title: string;
    amount: number;
    investment_id: number;
}

export interface CustomerInvestmentRes {
    id: number;
    title: string;
    amount: number;
    appreciation: number;
    created_at: string;
    investment_id: number;
    investment: InvestmentRes;
}

export interface CustomerInvestmentResOmitInvestment {
    id: number;
    title: string;
    amount: number;
    created_at: string;
}

export interface CustomerInvestmentWithdrawReq {
    id: number;
    amount: number;
}
